open-sphere-plugins/arcgis/src/main/java/io/opensphere/arcgis2/envoy/ArcGISLayerListEnvoy.java

```diff
@@ -166,20 +239,53 @@ public void query(DataModelCategory category, Collection<? extends Satisfaction>
         }
     }
 
+    /**
+     * Reads the response from the supplied URL from the supplied input stream.
+     * If debug is enabled, the response is logged before parsing.
+     *
+     * @param url the URL from which the response was received.
+     * @param is the input stream to read.
+     * @return a {@link FolderInfo} object containing the unmarshalled JSON
+     *         response.
+     * @throws IOException if the input stream cannot be read.
+     * @throws JsonParseException if the contents of the input stream cannot be
+     *             processed as JSON data.
+     * @throws JsonMappingException if the contents of the input stream cannot
+     *             be processed as JSON data.
+     */
+    private FolderInfo parseResponse(URL url, InputStream is) throws IOException, JsonParseException, JsonMappingException
+    {
+        FolderInfo info;
+        if (LOGGER.isDebugEnabled())
+        {
+            Charset charset = Charset.forName(System.getProperty("opensphere.charset", "UTF-8"));
+            String jsonResponse = IOUtils.toString(is, charset);
+            LOGGER.debug(url + ": " + jsonResponse);
+            info = JsonUtils.createMapper().readValue(jsonResponse, FolderInfo.class);
+        }
+        else
+        {
+            info = JsonUtils.createMapper().readValue(is, FolderInfo.class);
+        }
+        return info;
+    }
+
     /**
      * Start a query for a subfolder on the server.
      *
      * @param category The category for the current folder.
      * @param folderUrl The URL for the subfolder.
      * @param queryReceiver The receiver for the layers.
+     * @param includedSubpath the subpath to include in requests.
      * @throws QueryException If the query fails.
      * @throws InterruptedException If the thread is interrupted.
      */
-    private void queryFolder(DataModelCategory category, String folderUrl, CacheDepositReceiver queryReceiver)
+    private void queryFolder(DataModelCategory category, String folderUrl, CacheDepositReceiver queryReceiver,
+            String includedSubpath)
         throws InterruptedException, QueryException
     {
-        DataModelCategory nextDMC = category.withCategory(folderUrl);
-        query(nextDMC, null, null, null, 0, null, queryReceiver);
+        queryImpl(category.withCategory(folderUrl), queryReceiver,
+                UrlUtilities.toURL(category.withCategory(folderUrl).getCategory()), includedSubpath);
```

Comment:
Is not category.withCategory(folderUrl).getCategory() the same as folderUrl?

Reply:
Yep. I was inlining stuff, so it probably got pushed in there.